### PR TITLE
Feature/add special schema support

### PR DESCRIPTION
**Pull request by jmkerloch**
ref #55 

- update definition of table and schema name to use SQL formatting to get quoted values if needed (use of %I format option) (`internal/data/db_sql.go`)

- update return of notification for listener:
    - add use of %I SQL formatting to get quoted value for schema and table name
    - update `eventNotification` struct to get formatted Id

- fix creation and drop of trigger in listener to avoid issue with special schema and table name (`internal/data/listener.go`)

- add tests with special schema / table / column names

### CODE REVIEW

**Comment by benoitdm-oslandia** [comment]
You will do the same for table field names in this PR or in another one?

File: internal/data/listener.go [hunk 1]
```diff
@@ -46,6 +46,7 @@ type listenerDB struct {
 // event on the databases included in pg_featureserv. It is populated using the return value of
 // the pl/pgSQL procedure named `sqlNotifyFunction` defined in db_sql.go
 type eventNotification struct {
+	Id       string
```
Comment (benoitdm-oslandia) [inline internal/data/listener.go]: add comment

**Review by benoitdm-oslandia — CHANGES_REQUESTED** [review]

**Comment by jmkerloch** [comment]
> You will do the same for table field names in this PR or in another one?

I'm working on the update for table name and column name. This should be available soon.

File: internal/utiltest/test_db.go [hunk 2]
```diff
@@ -33,6 +33,10 @@ import (
 	log "github.com/sirupsen/logrus"
 )
 
+const SpecialSchemaStr = `"😀.$^{schema}.👿.😱"`
```
Comment (benoitdm-oslandia) [inline internal/utiltest/test_db.go]: what if you remove the `"`?

File: internal/utiltest/test_db.go [hunk 3]
```diff
@@ -154,21 +160,22 @@ func InsertSuperSimpleDataset(db *pgxpool.Pool, schema string) {
...
 	createBytes := []byte(`
 		DROP TABLE IF EXISTS %s CASCADE;
 		CREATE TABLE IF NOT EXISTS %s (
 			id SERIAL PRIMARY KEY,
-			geometry public.geometry(Point, 4326) NOT NULL
+			geometry public.geometry(Point, 4326) NOT NULL,
+			%s text
 		);
-		CREATE INDEX %s_geometry_idx ON %s USING GIST (geometry);
+		CREATE INDEX geometry_idx ON %s USING GIST (geometry);
```
Comment (benoitdm-oslandia) [inline internal/utiltest/test_db.go:176]: I do not know if it will work? Have the index names need to be different?

**Review by benoitdm-oslandia — CHANGES_REQUESTED** [review]

File: internal/utiltest/test_db.go (hunk 2 above)
Comment (jmkerloch) [reply internal/utiltest/test_db.go]: I you remove the `"` at the beginning and the end, everything will crash. To use special values you MUST use `"`.

File: internal/utiltest/test_db.go (hunk 3 above)
Comment (jmkerloch) [reply internal/utiltest/test_db.go:176]: We are creating index on different tables. index names must be different in the same table.

**Review by benoitdm-oslandia — APPROVED** [review]

**Pull request merged**